atomix/cluster/src/main/java/io/atomix/raft/storage/log/entry/UnserializedApplicationEntry.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.atomix.raft.storage.log.entry;
+
+import io.atomix.raft.storage.serializer.RaftEntrySerializer;
+import io.atomix.raft.storage.serializer.RaftEntrySerializer.SerializedBufferWriterAdapter;
+import io.camunda.zeebe.util.buffer.BufferWriter;
+
+/**
+ * An {@link ApplicationEntry} with unserialized application data. Used for writing new entries to
+ * the log.
+ */
+public record UnserializedApplicationEntry(
+    long lowestPosition, long highestPosition, BufferWriter writer) implements ApplicationEntry {
```

Comment:
:wrench: If you call the field `dataWriter`, you won't need the getter. Or am I missing something?